packages/@aws-cdk/aws-apigateway/lib/gateway-response.ts

```diff
@@ -0,0 +1,186 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnGatewayResponse } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+/**
+ * Properties for a new gateway response.
+ */
+export interface GatewayResponseProps extends GatewayResponseOptions {
+  /**
+   * Rest api resource to target.
+   */
+  readonly restApi: IRestApi;
+
+  /**
+   * Name of the gateway response.
+   * @default - Auto generated
+   */
+  readonly gatewayResponseName?: string;
+}
+
+/**
+ * Options to add gateway response.
+ */
+export interface GatewayResponseOptions {
+  /**
+   * Response type to associate with gateway response.
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+   */
+  readonly type: ResponseType;
+
+  /**
+   * Http status code for response.
+   * @default - standart http status code for the response type.
+   */
+  readonly statusCode?: string;
```

Comment:
Why not number?

Reply:
Because it is defined as string in most of the places i checked in cdk .
like : methodresponse.ts, integration.ts, application-listener-rules

Reply:
Not sure why these have been string here. @eladb - do you know?

Reply:
Let's switch this to a number, unless we hear otherwise.

Reply:
sorry for the confusion here. 
let's leave this as a string and keep it consistent with the rest of this code base.